core/store/src/trie/mod.rs

```diff
@@ -757,6 +760,97 @@ impl Trie {
     pub fn get_trie_nodes_count(&self) -> TrieNodesCount {
         self.storage.get_trie_nodes_count()
     }
+
+    /// Given a root and a key, it returns a MerkleProof for that key.
+    ///
+    ///  Whenever a proof is not found, an empty vector is returned.
+    ///
+    /// # Errors
+    ///
+    /// If the root is not found on the trie, a `StorageError` will be returned
+    pub fn get_proof(
+        &self,
+        root: &CryptoHash,
+        key: &[u8],
+    ) -> Result<Vec<TrieProofItem>, StorageError> {
+        let mut key = NibbleSlice::new(key);
+
+        let mut hash = *root;
+        if hash == Trie::empty_root() {
+            return Ok(vec![]);
+        }
+
+        let mut levels: Vec<TrieProofItem> = vec![];
+        loop {
+            let bytes = self.storage.retrieve_raw_bytes(&hash)?;
```

Comment:
nit: I wonder if it is worth it to add some local context when we return an error here.

Reply:
How do you suggest doing it? With a new error variant?

Reply:
Hmm... looking at the code, I don't think we can do much better here unless we do a big refactoring.  Sorry for the noise.